### PR TITLE
Rover: add dedicated sailboat paramiters

### PR DESCRIPTION
**Pull request by IamPete1**
This adds dedicated parameter for sailboat cross track error. Previously the way-point overshoot parameter was used. It also adds a new parameter for sailing loiter radius. 

These parameters can be added as now sailboat is a class there will be no cost to none-sailboat users. Additionally with the new motoring functionality sailboats can act as normal rovers requiring different parameters. Loiter radius is a good example, when sailing you need a large loiter radius as the vehicle must keep moving to maintain steering control. When using the motor such a large radius makes loiter mode pointless. Hence the addition of the new param.

### CODE REVIEW

**Comment by IamPete1** [comment]
this is the last bit of https://github.com/ArduPilot/ardupilot/pull/11423 except the sailboat throttle state waypoint which is waiting on https://github.com/ArduPilot/mavlink/pull/106

File: APMrover2/sailboat.cpp [hunk 1]
```diff
@@ -86,6 +86,24 @@ const AP_Param::GroupInfo Sailboat::var_info[] = {
     // @User: Standard
     AP_GROUPINFO("WNDSPD_MIN", 7, Sailboat, sail_windspeed_min, 0),
 
+    // @Param: SAIL_MAX_X_TRACK
```
Comment (rmackay9) [inline APMrover2/sailboat.cpp]: sorry to be annoying but could we rename this to SAIL_XTRACK_MAX or something similar.  Normally we put "_MAX" at the end of the parameter name.  Also I think we can drop the "SAIL_" part from the parameter description.

File: APMrover2/sailboat.cpp [hunk 2]
```diff
@@ -86,6 +86,24 @@ const AP_Param::GroupInfo Sailboat::var_info[] = {
...
+    // @DisplayName: Sailing vehicle max cross track error
+    // @Description: The sail boat will tack when it reaches this cross track error, defines a corridor of 2 times this value wide, 0 disables
+    // @Units: m
+    // @Range: 5 25
+    // @Increment: 1
+    // @User: Standard
+    AP_GROUPINFO("MAX_X_TRACK", 7, Sailboat, max_cross_track, 10),
+
+    // @Param: SAIL_LOIT_RADIUS
```
Comment (rmackay9) [inline APMrover2/sailboat.cpp]: we can leave off the "SAIL_" part of this parameter as well I think

File: APMrover2/sailboat.h [hunk 3]
```diff
@@ -96,6 +99,8 @@ class Sailboat
     AP_Float sail_heel_angle_max;
     AP_Float sail_no_go;
     AP_Float sail_windspeed_min;
+    AP_Float max_cross_track;
```
Comment (rmackay9) [inline APMrover2/sailboat.h]: let's put the max at the end of the variable name as well so let's call it cross_track_max or xtrack_max or something similar.

File: APMrover2/sailboat.cpp (hunk 1 above)
Comment (IamPete1) [reply APMrover2/sailboat.cpp]: fixed

File: APMrover2/sailboat.cpp (hunk 2 above)
Comment (IamPete1) [reply APMrover2/sailboat.cpp]: fixed

File: APMrover2/sailboat.h (hunk 3 above)
Comment (IamPete1) [reply APMrover2/sailboat.h]: fixed

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**